app/move/app/new/steps.js

```diff
@@ -137,14 +138,28 @@ module.exports = {
         value: true,
         next: 'personal-details',
       },
+      {
+        fn: Base.prototype.shouldAskYouthSentenceStep,
+        next: FEATURE_FLAGS.YOUTH_RISK_ASSESSMENT
+          ? 'serving-youth-sentence'
+          : 'move-details',
+      },
       'move-details',
     ],
     fields: ['people'],
   },
   '/personal-details': {
     controller: PersonalDetails,
     pageTitle: 'moves::steps.personal_details.heading',
-    next: 'move-details',
+    next: [
+      {
+        fn: Base.prototype.shouldAskYouthSentenceStep,
```

Comment:
Could we not just set a value in the journey based on the feature flag value and simply have a corresponding next step for the 2 possibilities?

Reply:
I was struggling where to do it really, and I didn't want to set that value on every save. 

The logic is a little more complicated than the value submitted by a form and I felt this was the neatest way to extract that logic.

By using a method it means it will not be run with every request. 

If we were to set a value during the "save" it would also need to be done both via the person search results and creating a new person and that then felt a little harder to follow.